Dockerfile

```diff
@@ -1,52 +1,53 @@
-# Build the manager binary
-FROM golang:1.22 as builder
+FROM nixos/nix:latest AS go-builder
+COPY Makefile Makefile
+COPY shell.nix shell.nix
 
+# Set the working directory
 WORKDIR /workspace
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
-RUN go install golang.org/x/tools/cmd/goimports@latest
-
-# Copy the go source & git info
-COPY cmd/manager/main.go cmd/manager/main.go
-COPY .git/ .git/
-COPY pkg/ pkg/
-COPY internal/ internal/
-COPY Makefile Makefile
-COPY config/ config/
-COPY hack/ hack/
 
+# Copy necessary files
+COPY . .
```

Comment:
... and remove this one. It might copy things we do not need or want to copy in Docker.